libraries/cbor/src/macros.rs

```diff
@@ -41,20 +41,31 @@ macro_rules! cbor_map_options {
     };
 
     ( $( $key:expr => $value:expr ),* ) => {
+        {
+            let mut _map = ::alloc::collections::BTreeMap::<_, $crate::values::Value>::new();
+            extend_cbor_map_options! ( &mut _map, $( $key => $value, )* );
+            $crate::values::Value::Map(_map)
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! extend_cbor_map_options {
+    // Add trailing comma if missing.
+    ( &mut $map:expr, $( $key:expr => $value:expr ),+ ) => {
```

Comment:
The syntax shouldn't require a `&mut` token. The caller could obtain a mutable reference without directly using `&mut`, for example by calling a function which returns a `&mut T`.

Reply:
Forget about my other comment. I didn't mean `&mut` as literally it in the source code, but rather as an expression of type `&mut T`. It doesn't make sense to put `&mut` in the macro's syntax.

Reply:
Obsolete.